client/views/omnichannel/currentChats/FilterByText.tsx

```diff
@@ -74,11 +76,46 @@ const FilterByText: FilterByTextType = ({ setFilter, reload, ...props }) => {
 	const Tags = useCurrentChatTags();
 
 	const onSubmit = useMutableCallback((e) => e.preventDefault());
-	const reducer = function (acc: any, curr: string): any {
-		acc[curr] = '';
-		return acc;
+
+	const handleCustomFieldsFilterToggle = (): void => setCustomFieldsSearchBarOpen(true);
+
+	const { value: allCustomFieldsDefinition, phase: stateCustomFields } = useEndpointData('livechat/custom-fields');
+
+	const jsonConverterToValidFormat = (customFields: Omit<ILivechatCustomField, '_updatedAt'>[]): CustomFieldsData => {
+		const jsonObj: CustomFieldsData = {};
+		customFields.forEach(({ _id, label, visibility, options, scope, required }) => {
+			if (visibility === 'visible' && scope === 'room') {
+				const optionsArrayWithClearFilterOption = options
+					? options
+							.split(',')
+							.map((item) => item.trim())
+							.filter((item) => item)
+					: [];
+				optionsArrayWithClearFilterOption.length && optionsArrayWithClearFilterOption.unshift(t('None_no_option_selected'));
+				jsonObj[_id] = {
+					label,
+					type: options ? 'select' : 'text',
+					required,
+					defaultValue: options ? t('None_no_option_selected') : '',
+					...(optionsArrayWithClearFilterOption.length && {
+						options: optionsArrayWithClearFilterOption,
+					}),
+				};
+			}
+		});
+		return jsonObj;
 	};
 
+	const jsonCustomField = useMemo(
+		() => (allCustomFieldsDefinition?.customFields ? jsonConverterToValidFormat(allCustomFieldsDefinition.customFields) : {}),
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I was getting an eslint warning about adding `jsonConverterToValidFormat` to this dependency array, hence I'd added this before. However now I've removed it since you recommended moving that function outside of this component, so this won't be required